docs/reference/navigation.rst

```diff
@@ -250,9 +254,150 @@ If true this node is a :ref:`soft root <soft-root>`. A page can be marked as a *
 in its 'Advanced Settings'.
 
 
-
 ******************************
 Modifying & Extending the menu
 ******************************
 
 Please refer to the :doc:`/how_to/menus` documentation
+
+
+********************************
+Menu system classes and function
+********************************
+
+``menu`` application
+====================
+
+..  class:: menus.base.Menu
+
+    The base class for all menu-generating classes.
+
+    ..  method:: get_nodes(self, request)
+
+        Each sub-class of ``Menu`` should return a list of NavigationNode instances.
+
+
+..  class:: menus.base.Modifier
+
+    The base class for all menu-modifying classes. A modifier add, removes or changes NavigationNodes in the list.
+
+    ..  method:: modify(self, request, nodes, namespace, root_id, post_cut, breadcrumb)
+
+        Each sub-class of ``Modifier`` should implement a ``modify()`` method.
+
+
+..  class:: menus.menu_pool.MenuPool
+
+    ..  method:: get_nodes()
```

Comment:
This was deprecated on 3.3 and replaced with `menu_pool.get_renderer(request).get_nodes()`